lib/Backends/Interpreter/InterpreterDeviceManager.cpp

```diff
@@ -46,6 +57,44 @@ uint64_t InterpreterDeviceManager::getAvailableMemory() const {
   return maxMemoryBytes_ - usedMemoryBytes_;
 }
 
+llvm::Expected<uintptr_t>
+InterpreterDeviceManager::getPlaceholderAddress(std::string function,
+                                                const Placeholder *P) {
+  // Peer-to-peer communication is implemented in the Interpreter backend using
+  // future-promise pairs (refered to a as a tensor pipe). Each (functionName,
+  // placeholder) pair maps to a tensor pipe that can be used to send/receive a
+  // Tensor.
+
+  // Convert P to uintptr_t since that's what's stored in the map.
+  uintptr_t addr = reinterpret_cast<uintptr_t>(P);
```

Comment:
May be you can consider to use incremental atomic counter instead? Addressed can be reused by compiler. Once memory got released new allocation can grab it and in case of race conditions wrong entry can be referred.